src/ucp/api/ucp.h

```diff
@@ -1497,9 +1525,37 @@ struct ucp_am_recv_param {
     uint64_t           recv_attr;
 
     /**
-     * Endpoint, which can be used for reply to this message.
+     * Endpoint, which can be used for the reply to this message.
      */
     ucp_ep_h           reply_ep;
+
+    /**
+     * Length of the whole message in bytes. Relevant for multi-fragment eager
+     * messages handled by data handlers registered without UCP_AM_FLAG_WHOLE_MSG
+     * flag.
+     */
+    size_t             total_length;
+
+    /**
+     * Offset of the message fragment in bytes. Layout of the multi-fragment
```

Comment:
The offset of the message fragment in bytes relative to the beginning of overall message ?  Where is fragment size ?

Reply:
description of all other fields start without `the`, i'd avoid it here as well.
The fragment size is passed as length to the data callback (described in ucp_def.h)